EarlGrey/Additions/NSURLSession+GREYAdditions.m

```diff
@@ -62,6 +62,10 @@ - (NSURLSessionDataTask *)greyswizzled_dataTaskWithRequest:(NSURLRequest *)reque
     // request need not be tracked as its completion/failure does not trigger any delegate
     // callbacks.
     if ([delegate respondsToSelector:originalSel]) {
+      // This is to solve issues where there is a proxy delegate instead of the original instance (e.g. TrustKit).
```

Comment:
You'd need a check in line 59 as well otherwise it will check wrong delegate class for existence of swizzled selector. I feel like there's a cleaner way to write up when and how to swizzle logic:
1) change line 59 to check if delegate responds to swizzledSel
2) inside `@synchronized(selfClass)` first check if [delegateClass instancesRespondToSelector:original], if not, get the forwarding target becase according to https://developer.apple.com/documentation/objectivec/nsobject/1418555-instancesrespondtoselector?language=objc "If aSelector messages are forwarded to other objects, instances of the class are able to receive those messages without error even though this method returns NO." so you should be able to leverage that and grab the right class to swizzle. Now if that class forwards it again...well let's just worry about it if we run into that case :)

Reply:
Changing 59 is not necessary. It's what the system asks when deciding which protocol methods are implemented. The documentation might be correct in theory regarding the ObjC runtime, but the way Apple handles delegates is by keeping a packed struct of delegate capabilities, and only calls those delegate methods iff the cap flag of the required method is enabled. The cap flags are set once upon delegate setting. You can verify this by overriding `respondsToSelector:` and printing the selectors. You will notice that the method is called in bulk for each delegate method at the time of setting the delegate.